docsrc/content/computation-expressions.fsx

```diff
@@ -31,11 +31,26 @@ There is a single computation expression: ``monad`` but it comes in 4 flavours:
 
 In other words:
 
- - ``monad.fx`` is the same as ``monad``: Lazy monadic builder. Use when you want to use side-effects instead of the additive behavior of monad plus.
- - ``monad.fx'`` is the strict version of ``monad``
- - ``monad.plus`` Lazy additive monadic builder. Use when you expect one or more results.
+ - ``monad.fx`` or simply ``monad``: Lazy monadic builder. Use when you want to use side-effects instead of the additive behavior of monad plus.
+ - ``monad.fx.strict`` (or ``monad.fx'`` or simply ``monad.strict`` or ``monad'``) is the strict version of ``monad``.
+ - ``monad.plus``: Lazy additive monadic builder. Use when you expect one or more results.
  - ``monad.plus'`` is the strict version of ``monad.plus``
 
+Note that a type is either lazy or strict, but it could act as fx or plus at the same time (see below some examples). This means that we need to pay attention when using a CE over a type, if the type is lazy but with use a strict monad, we'll get strict semantics which probably would make no sense, but if we do the opposite we might run into runtime errors, fortunately a compile-time warning (or error) will prevent us.
+
+A simple way to find out if a type is strict or lazy is to execute this in fsi: `let _ : MyType<'t> = monad { printfn "I'm strict" }`
+
+For layered monads (monad transformers) the general rule is: the monad is strict unless at least one of its constituent types is lazy, in that case the whole monad becomes lazy.
+
+*)
+
+let _ : OptionT<list<unit option>> = monad { printfn "I'm strict" }
```

Comment:
I think that you need to import `FSharpPlus.Data`, while also include the reference to the F#+ dll (done in this file a bit further down)

Reply:
Yes, maybe I can move the open declarations further up, so those one liners compile.